<h1>Assignment 7: Aggregating Data</h1>

In [30]:
import pandas as pd
import numpy as np
import os
os.chdir('/Users/mars/Downloads/Local/')
Movies = pd.read_csv('Movies.csv',index_col=0 )


<p>First, I am replacing null values with 0 to make aggregations easier.</p>

In [31]:
Movies.fillna(0)

I Care a Lot  Love & Basketball  Jackie Brown  Bird Box  Hereditary
Piankhy           2.0                5.0           5.0       2.5         4.5
Marissa           1.5                4.0           4.0       3.0         5.0
Shannon           3.0                0.0           0.0       3.5         2.0
Jose              4.0                3.5           4.0       2.5         3.0
Jane              3.5                0.0           0.0       2.0         5.0

Here, we will find the average rating for each person.

In [32]:
Movies.mean(axis=1)


Piankhy    3.800000
Marissa    3.500000
Shannon    2.833333
Jose       3.400000
Jane       3.500000
dtype: float64

<p>The results are:<br>
    <b>Piankhy</b> = 3.8<br>
    <b>Marissa</b> = 3.5<br>
    <b>Shannon</b> = 2.8<br>
    <b>Jose</b> = 3.4<br>
    <b>Jane</b> = 3.5</p>

Next, we find the average rating for each movie.

In [33]:
Movies.mean(axis=0)

I Care a Lot         2.800000
Love & Basketball    4.166667
Jackie Brown         4.333333
Bird Box             2.700000
Hereditary           3.900000
dtype: float64

The results are:<br>
    <b>I Care a Lot</b>: 2.8<br>
    <b>Love & Basketball</b>: 4.2<br>
    <b>Jackie Brown</b>: 4.3<br>
    <b>Bird Box</b>: 2.7<br>
    <b>Hereditary</b>: 3.9<br>

<h3>Part 2: Normalize rating data</h3>
<p>This time, we will get the <i>normalized</i> averages for each person and each movie.</p>

In [40]:
Movies_n=((Movies-Movies.min())/(Movies.max()-Movies.min()))
print(Movies_n)

Movies_n=((Movies-Movies.min())/(Movies.max()-Movies.min()))
Movies_n.mean(axis=0)

         I Care a Lot  Love & Basketball  Jackie Brown  Bird Box  Hereditary
Piankhy           0.2           1.000000           1.0  0.333333    0.833333
Marissa           0.0           0.333333           0.0  0.666667    1.000000
Shannon           0.6                NaN           NaN  1.000000    0.000000
Jose              1.0           0.000000           0.0  0.333333    0.333333
Jane              0.8                NaN           NaN  0.000000    1.000000


I Care a Lot         0.520000
Love & Basketball    0.444444
Jackie Brown         0.333333
Bird Box             0.466667
Hereditary           0.633333
dtype: float64

<p>The normalized averages for each movie are:<br>
    <b>I Care a Lot:</b>.52<br>
    <b>Love & Basketball:</b>.44<br> 
    <b>Jackie Brown:</b>.33<br>
    <b>Bird Box:</b>.46<br>
    <b>Hereditary:</b>.63<br></p>
        

In [41]:
Movies_n.mean(axis=1)

Piankhy    0.673333
Marissa    0.400000
Shannon    0.533333
Jose       0.333333
Jane       0.600000
dtype: float64

<p>The normalized averages by person are:<br>
    <b>Piankhy:</b>0.67<br>
    <b>Marissa:</b>0.40<br>
    <b>Shannon:</b>0.53<br>
    <b>Jose:</b>0.33<br> 
    <b>Jane:</b>0.60<br></p>
 <br><br>
    
    
<p>In this example, normalizing the data isn't super useful, since usually normalization is used to scale data that may be in different units or to scale outliers. This data didn't really have outliers since there is so little data and it is on such a small scale (1-5). Perhaps normalizing would be useful if the ratings were done on a scale of 1-10 and most ratings were under 5, leaving some data far out of the 'normal' range. </p>